GeneralsMD/Code/GameEngine/Source/GameLogic/System/GameLogic.cpp

```diff
@@ -2834,14 +2834,17 @@ Int GameLogic::rebalanceChildSleepyUpdate(Int i)
 	UpdateModulePtr* pI = &m_sleepyUpdates[i];
 
 	// our children are i*2 and i*2+1
-  Int child = ((i+1)<<1)-1;
+	Int child = ((i+1)<<1)-1;
+	UnsignedInt containerSize = m_sleepyUpdates.size();
+	// TheSuperHackers @fix Mauller 02/04/2025 Prevent invalid index into container throwing errors during debug
+	if(child >= containerSize) return i;
 	UpdateModulePtr* pChild = &m_sleepyUpdates[child];
-	UpdateModulePtr* pSZ = &m_sleepyUpdates[m_sleepyUpdates.size()];	// yes, this is off the end.
+	UpdateModulePtr* pSZ = &m_sleepyUpdates[containerSize - 1];
```

Comment:
I think the way this was meant to be written is
```cpp
UpdateModulePtr* pSZ = &m_sleepyUpdates + m_sleepyUpdates.size();
```
Then all other changes here can be reverted.